### ISSUE TITLE
Jest Snapshot Testing Issues

### ISSUE THREAD

**Issue by chufnagel** [planttheidea/react-pure-lifecycle#7]
Any advice on getting this this to work with Jest Snapshots? My teammate added this to a functional component I wrote:
```javascript
const methods =  {
  componentDidMount(props) {
    console.log('props',props)
    props.setNavFlagToDashboard()
  }
}

export default lifecycle(methods)(FunctionalComponent)
```

Which leads to this output when running the test:
`TypeError: props.setNavFlagToDashboard is not a function`

Any advice on how best to approach this?

**Comment by planttheidea** [comment]
It's hard to sufficiently answer that question without seeing the test itself. Can you provide the test code as well?

**Comment by chufnagel** [comment]
I realize this is more attributable to me being new-ish to Enzyme/Jest/React than anything - I'm sorry for the trouble!

Here's the test in question:
```javascript
test("BlogList renders without crashing", () => {
  const component = shallow(<BlogList />);
  expect(component.exists()).toEqual(true);
});
test("BlogList renders a blog post", () => {
  const component = shallow(
    <BlogList blogs={testBlogs} />
  );
  expect(component).toMatchSnapshot();
});
```

**Comment by planttheidea** [comment]
Well based on this it simply looks like you're not passing the function prop in question to the component. Have you tried just passing in a noop function?

`shallow(<BlogList setNavFlagToDashboard={() => {}} />)`

**Comment by chufnagel** [comment]
Yes, no luck though. I'll dig into it more when this project is over - thanks for taking a look!

**Comment by planttheidea** [comment]
No problem, I'll try to dust this project off, I haven't touched it in a while. Since I released `react-parm` (which accomplishes the same goal plus a whole lot more) I haven't been as active with this repo.

**Comment by planttheidea** [comment]
So I just released `3.0.0`, but that was just a refresh of the codebase for modern approach ... I didn't actually change anything functionally, so updating should have no impact on your codebase (except tree-shaking is now active, since you use webpack :smiley_cat: ).

I also added a test to try to reproduce, and couldn't, so hopefully this example provides some guidance:

```javascript
test('if snapshots work as expected', (t) => {
  const onMount = ({doThing}) => doThing();
  const spy = sinon.spy();

  const DummyComponent = componentDidMount(onMount)(FunctionalComponent);

  const wrapper = shallow(<DummyComponent doThing={spy} />);

  t.true(spy.calledOnce);

  t.snapshot(toJson(wrapper));
});
```

This is with AVA, not Jest, but that should not make a difference ... likely it was in the way you were applying things.
